index.js

```diff
@@ -485,10 +470,28 @@ function flushSync (stream) {
     }
 
     if (++spins === 10) {
-      throw new Error('_flushSync took too long (10s)')
+      throw new Error('waitSync took too long (10s)')
     }
   }
-  // process._rawDebug('flushSync finished')
+  // process._rawDebug('waitSync finished')
+}
+
+function waitAsync (stream, cb) {
```

Comment:
There is an implementation of this in lib.

Reply:
I just moved the existing code here. This is not something new.